presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ReorderJoinsUtils.java

```diff
@@ -142,41 +148,36 @@ else if (rightSymbols.containsAll(filterSymbols)) {
                 Optional.of(PARTITIONED));
     }
 
-    public static PriorityQueue<JoinNode> planPriorityQueue(SymbolAllocator symbolAllocator, Lookup lookup, Session session, CostComparator costComparator)
+    private static List<JoinNode.EquiJoinClause> flipJoinCriteriaIfNecessary(List<JoinNode.EquiJoinClause> joinCriteria, PlanNode left)
     {
-        return new PriorityQueue<>((node1, node2) -> {
-            PlanNodeCostEstimate node1Cost = lookup.getCumulativeCost(session, symbolAllocator.getTypes(), node1);
-            PlanNodeCostEstimate node2Cost = lookup.getCumulativeCost(session, symbolAllocator.getTypes(), node2);
-            return costComparator.compare(session, node1Cost, node2Cost);
-        });
+        return joinCriteria
+                .stream()
+                .map(x -> left.getOutputSymbols().contains(x.getLeft()) ? x : new JoinNode.EquiJoinClause(x.getRight(), x.getLeft()))
+                .collect(toImmutableList());
     }
 
-    public static JoinNode flipJoin(JoinNode joinNode)
+    private static PlanNode getJoinSource(PlanNodeIdAllocator idAllocator, List<PlanNode> leftNodes, List<Expression> leftFilters, List<JoinNode.EquiJoinClause> leftCriteria)
```

Comment:
Why are the arguments here phrased in terms of the left? Isn't this method generic for any set of nodes, filters and criteria?